### PR TITLE
Public Stuff + fixing

### PR DESCRIPTION
**Pull request by oakeyc**

### CODE REVIEW

File: azclishell/app.py [hunk 1]
```diff
@@ -402,6 +402,9 @@ def _special_cases(self, text, cmd, outside):
                 os.path.join(
                     SHELL_CONFIGURATION.get_config_dir(),
                     SHELL_CONFIGURATION.get_history())
+        if '--version' in text:
+            cmd = 'az ' + cmd
+            outside = True
```
Comment (derekbekoe) [inline azclishell/app.py]: `outside` would run the CLI in a separate process/shell right? Seems hacky...

File: azclishell/app.py [hunk 2]
```diff
@@ -452,6 +455,7 @@ def _special_cases(self, text, cmd, outside):
                 set_default_command(self.default_command, add=False)
                 print('undefaulting: ' + value[0])
             cmd = cmd.replace(SELECT_SYMBOL['undefault'], '')
+            c_flag = True
```
Comment (derekbekoe) [inline azclishell/app.py]: What does this do?

Comment (oakeyc) [inline azclishell/app.py]: tells the loop to go on to the next iteration without executing

File: azclishell/app.py (hunk 1 above)
Comment (oakeyc) [inline azclishell/app.py]: Yeah, but calling the method to do it does a system exit, and copy and pasting the same code wouldn't work if things changed in the cli

**Pull request merged**